hack/golint.sh

```diff
@@ -15,6 +15,6 @@ else
     --volume "${PWD}:/go/src/github.com/metal3-io/baremetal-operator:ro,z" \
     --entrypoint sh \
     --workdir /go/src/github.com/metal3-io/baremetal-operator \
-    quay.io/metal3-io/golint:latest \
+    docker.io/golangci/golangci-lint:latest \
```

Comment:
quay.io/metal3-io/golint:latest is the one used in CI, so if we don't use that then potentially developers will get different results to what CI gets.

Reply:
After your and @honza's comments, I realized that this is not simply changeable image. It is better to stay using `quay.io/metal3-io/golint:latest` and make changes on it, if it is necessary.